lib/Analysis/PtrUseVisitor.cpp

```diff
@@ -35,11 +35,5 @@ bool detail::PtrUseVisitorBase::adjustOffsetForGEP(GetElementPtrInst &GEPI) {
   if (!IsOffsetKnown)
     return false;
 
-  // GEP pointer operand may be address space casted from the original pointer
-  // therefore the expected offset size need to match the pointer operand.
```

Comment:
@yxsamliu IIRC this was here to catch the case where the address and the offset have different widths, do you know if this is still needed?

Reply:
@yxsamliu could you take a quick look at this? thanks

Reply:
@david-salinas any particular reason why this needs to be removed?

Reply:
I believe leaving this code in prevents an LLVM lit test from passing - though I don't recall which one.  No HCC/HIP tests fail when it is removed.

Reply:
Could you create a new PR with only this change in llvm and then update the hcc config with this change only(don't want the next auto-merge to pick up this)?
Also I think this PR should be closed since it's obsoleted

Reply:
The other changes are already in LLVM?  OK.  We can close this PR then.

Reply:
I'm not sure, there were too many changes in this PR, should do another round and find out the deltas between this and amd-common.